### ISSUE TITLE
WebSocket upgrade request missing the Connection header

### ISSUE THREAD

**Issue by 4e6** [helidon-io/helidon#8189]
[//]: # "Please remove these comments"
[//]: # "Provide a general summary of the issue in the Title above"

## Environment Details
* Helidon Version: 4.0.2
* Helidon SE 
* JDK version: openjdk 21.0.1
* OS: Archlinux
* Docker version (if applicable): -

----------

## Problem Description
[//]: # "Describe the bug in detail highlighting current behavior vs expected behavior"
[//]: # "State if the problem is easily reproducible or happens intermittently"
[//]: # "Include stack traces or command outputs"

When trying to create a WebSocket connection with `WsClient`, it fails to upgrade the connection

```
Exception in thread "main" io.helidon.webclient.websocket.WsClientException: Failed to upgrade to WebSocket. Response: 200 OK: Via: 1.1 fly.io
Content-Length: 630
Content-Type: text/plain
Date: Wed, 03 Jan 2024 19:57:01 GMT
Server: Fly/ec8196c09 (2024-01-02)
fly-request-id: 01HK8ED5H2PDAM0NC27TD3ZGQD-lhr

	at io.helidon.webclient.websocket@4.0.2/io.helicon.webclient.websocket.WsClientImpl.connect(WsClientImpl.java:123)
```

According to the [Mozilla doc](https://developer.mozilla.org/en-US/docs/Web/HTTP/Protocol_upgrade_mechanism#upgrading_to_a_websocket_connection), a WebSocket upgrade request should contain two headers:
```
Connection: Upgrade
Upgrade: websocket
```
Debugging showed that the upgrade request sets the `Upgrade` header but is missing the `Connection` one https://github.com/helidon-io/helidon/blob/68de0bb538483c5f76fdf78c686e273f15b4c684/webclient/http1/src/main/java/io/helidon/webclient/http1/Http1ClientRequestImpl.java#L130-L133

After setting the `Connection` header (see the example below), the connection was successfully established.

## Steps to reproduce
[//]: # "Step by step instructions to reproduce the problem"
[//]: # "Provide sample code/application if relevant"

This example results in an `io.helidon.webclient.websocket.WsClientException: Failed to upgrade to WebSocket.`

```
io.helidon.webclient.websocket.WsClient.builder()
    .build()
    .connect(
        new java.net.URI("wss://echo.websocket.org/"), 
        new io.helidon.websocket.WsListener() {}
    );
 ```

After adding a `Connection` header, the connection is established successfully

```
var headersMap = new java.util.HashMap<String, String>();
headersMap.put("CONNECTION", "Upgrade");
io.helidon.webclient.websocket.WsClient.builder()
    .defaultHeadersMap(headersMap)
    .build()
    .connect(
        new java.net.URI("wss://echo.websocket.org/"), 
        new io.helidon.websocket.WsListener() {}
    );
```

**Comment by spericas** [comment]
@4e6 Thanks for the report, yes it looks like that header is missing. Just to fully understand your scenario, you are using Helidon as a client of some third-party WS server right?

**Comment by 4e6** [comment]
Yes, I would like to establish a connection with the Yjs server [yjs/y-websocket](https://github.com/yjs/y-websocket).
